v2/api/containerinstance/v1beta20211001/container_group_types_gen.go

```diff
@@ -7783,6 +7548,190 @@ const (
 	IpAddress_Type_STATUS_Public  = IpAddress_Type_STATUS("Public")
 )
 
+// Generated from: https://schema.management.azure.com/schemas/2021-10-01/Microsoft.ContainerInstance.json#/definitions/LogAnalytics
+type LogAnalytics struct {
+	// LogType: The log type to be used.
+	LogType *LogAnalytics_LogType `json:"logType,omitempty"`
+
+	// Metadata: Metadata for log analytics.
+	Metadata map[string]string `json:"metadata,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// WorkspaceId: The workspace id for log analytics
+	WorkspaceId *string `json:"workspaceId,omitempty"`
+
+	// +kubebuilder:validation:Required
+	// WorkspaceKey: The workspace key for log analytics
+	WorkspaceKey *string `json:"workspaceKey,omitempty"`
```

Comment:
I think this should be a secret

Reply:
Hmm, it's flagged as `x-ms-secret` in the Swagger too... see [here](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2021-10-01/containerInstance.json#L1560)

```
        "workspaceKey": {
          "description": "The workspace key for log analytics",
          "type": "string",
          "x-ms-secret": true
        },
```